From 358be776e628b408192a88d6ea3e68a890d32a39 Mon Sep 17 00:00:00 2001
From: Juan Gutierrez <juan.gutierrez@nxp.com>
Date: Wed, 13 Apr 2016 17:35:09 -0500
Subject: [PATCH 05/23] dtb: imx6sxscm: adding lcd suport for imx6sxscm

A new dts file imx6sxscm-1gb-evb-ldo-lcdif1.dts is created for
supporting lcd configuration.

setenv panel 'MCIMX28LCD'
setenv fdt_file 'imx6sxscm-1gb-evb-ldo-lcdif1.dtb'
setenv mmcargs 'setenv bootargs console=${console},${baudrate} \
                root=${mmcroot} video=mxc_lcdif:SEIKO-WVGA,bpp=16'

Signed-off-by: Juan Gutierrez <juan.gutierrez@nxp.com>
---
 arch/arm/boot/dts/Makefile                         |  1 +
 arch/arm/boot/dts/imx6sxscm-1gb-evb-ldo-lcdif1.dts | 36 ++++++++++++++++++++++
 2 files changed, 37 insertions(+)
 create mode 100644 arch/arm/boot/dts/imx6sxscm-1gb-evb-ldo-lcdif1.dts

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index 7a25426..dbde3af 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -243,6 +243,7 @@ dtb-$(CONFIG_ARCH_MXC) += \
 	imx6sx-sdb-emmc.dtb \
 	imx6sx-sdb-m4.dtb \
 	imx6sxscm-1gb-evb-ldo.dtb \
+	imx6sxscm-1gb-evb-ldo-lcdif1.dtb \
 	vf610-twr.dtb \
 	imx7d-12x12-lpddr3-arm2.dtb \
 	imx7d-12x12-ddr3-arm2.dtb \
diff --git a/arch/arm/boot/dts/imx6sxscm-1gb-evb-ldo-lcdif1.dts b/arch/arm/boot/dts/imx6sxscm-1gb-evb-ldo-lcdif1.dts
new file mode 100644
index 0000000..fd6eaf1
--- /dev/null
+++ b/arch/arm/boot/dts/imx6sxscm-1gb-evb-ldo-lcdif1.dts
@@ -0,0 +1,36 @@
+/*
+ * Copyright (C) 2014-2015 Freescale Semiconductor, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+#include "imx6sx-sdb-ldo.dts"
+#include "imx6sxscm-evb.dtsi"
+
+/ {
+	regulators {
+		reg_lcd_3v3: lcd-3v3 {
+			status = "okay";
+		};
+	};
+
+	sii902x_reset: sii902x-reset {
+		status = "okay";
+	};
+};
+
+&csi1 {
+	status = "disabled";
+};
+
+&lcdif1 {
+	status = "okay";
+};
+
+&i2c1 {
+	sii902x@39 {
+		status = "okay";
+	};
+};
-- 
1.9.1

